File: python/substr_with_vowels.py
"""
Number of lowercase 3 character length sub strings with exactly 2 vowels
"""

vowels = ["a", "e", "i", "o", "u"]


def solution(text: str) -> int:
    count = 0
    num_groups = len(text) // 3
    start = 0

    for _ in range(num_groups):
        end = start + 3
        sub_str = text[start:end]
        num = len([ch for ch in sub_str if ch in vowels])
        if num == 2:
            count += 1
        start = end
    return count


if __name__ == '__main__':
    assert(solution("aobide") == 2)
    assert(solution("aobidetyuityorevghtyet") == 3)

File: go/substr_with_vowels.go
package main

import "fmt"

const vowels = "aeoiu"

func check_vowels(char byte) bool {
	result := false
	for i := 0; i < len(vowels); i++ {
		if vowels[i] == char {
			result = true
			break
		}
	}
	return result
}

func solution(text string) int {
	var number = 0
	var count = 0

	var num_groups int = len(text) / 3
	var start = 0
	var end = 0

	for ngroup := 0; ngroup <= num_groups; ngroup++ {
		end = start + 3
		if end > len(text) {
			end = len(text)
		}
		sub_str := text[start:end]
		for _, ch := range sub_str {
			if check_vowels(byte(ch)) {
				number += 1
			}
		}

		if number == 2 {
			count += 1
		}

		number = 0
		start = end
	}

	return count
}

func main() {
	result := solution("aibebo")
	result_1 := solution("aobidetyuityorevghtyet")

	fmt.Printf("Result: %d\n", result)
	fmt.Printf("Result 1: %d\n", result_1)
}
